internal/state/module.go

```diff
@@ -465,7 +503,13 @@ func (s *ModuleStore) UpdateDiagnostics(path string, diags map[string]hcl.Diagno
 		return err
 	}
 
-	mod.Diagnostics = diags
+	langID, _ := lsp.LanguageIDFromPath(path)
```

Comment:
Similarly here I assumed we would have a dedicated method, e.g.

```go
func (s *ModuleStore) UpdateVariableDiagnostics(path string, diags map[string]hcl.Diagnostics) error
```

which would then negate the need for the conditional.